feat: Design Token (#1107)
AmadeusITGroup/otter: packages/@o3r/design/README.md

> <h1 align="center">Otter Design</h1>

Reviewer: Some documentation on how to use and what kind of code it generates would be nice

Author: Created the task #1126 to follow it.

<p align="center">
  <img src="https://raw.githubusercontent.com/AmadeusITGroup/otter/main/assets/logo/otter.png" alt="Super cute Otter!" width="40%"/>
</p>

This package is an [Otter Framework Module](https://github.com/AmadeusITGroup/otter/tree/main/docs/core/MODULE.md).
<br />
<br />

## Description

> Set of tools to generate CSS themes and [Metadata](https://github.com/AmadeusITGroup/otter/tree/main/docs/cms-adapters/CMS_ADAPTERS) based on [Design Token Specification](https://design-tokens.github.io/community-group/format/).

Reviewer: ```suggestion
Set of tools to generate CSS themes and [Metadata](https://github.com/AmadeusITGroup/otter/tree/main/docs/cms-adapters/CMS_ADAPTERS) based on the [Design Token Specifications](https://design-tokens.github.io/community-group/format/).
```

ng add @o3r/design
```

## Generators

Otter Design module provides a set of code generators based on [angular schematics](https://angular.io/guide/schematics).

| Schematics   | Description                                             | How to use           |
| ------------ | ------------------------------------------------------- | -------------------- |
| add          | Include Otter design module in a library / application. | `ng add @o3r/design` |
| generate-css | Generate CSS Theme based on Design Token Files          | `ng g generate-css`  |

## Builders

Otter Design module provides a set of builders based on [angular builders](https://angular.io/guide/cli-builder).

### generate-css

The `generate-css` builder can generate CSS and CMS Metadata based on given Design Token Json files.
The following configurations are available:

| Options                     | Default Value  | Description                                                                                                                 |
| --------------------------- | -------------- | --------------------------------------------------------------------------------------------------------------------------- |

> | **designTokenFilePatterns** | [] *Require*   | Path patterns to the Design Token JSON files. <br /> File in dependencies are supported and resolved with Node Resolver.    |

Reviewer: ```suggestion
| **designTokenFilePatterns** | [] *Require*   | Path patterns to the Design Token JSON files. <br /> Files in dependencies are supported and resolved with Node Resolver.    |
```

> | **output**                  | *null*         | Output file where generate the CSS. <br /> The path specified in `o3rTargetFile` will be ignore if this option is specified |

Reviewer: ```suggestion
| **output**                  | *null*         | Output file for the CSS generation. <br /> The path specified in `o3rTargetFile` will be ignore if this option is specified |
```

> | **defaultStyleFile**        | src/theme.scss | File path to generate the variable if not determined by the specification                                                   |

Reviewer: ```suggestion
| **defaultStyleFile**        | src/theme.scss | File path to generate the variable if not determined by the specification  s                                                 |
```

| **metadataOutput**          | *null*         | Path to generate the metadata for CMS. <br /> The Metadata will be generated only if the file path is specified.            |
| **rootPath**                | *null*         | Root path of files where the CSS will be generated.                                                                         |
| **failOnDuplicate**         | false          | Determine if the process should stop in case of Token duplication.                                                          |

> | **prefix**                  | *null*         | Prefix to happen to generated variables.                                                                                    |

Reviewer: ```suggestion
| **prefix**                  | *null*         | Prefix to append to generated variables.                                                                                    |
```
